docker/Dockerfile

```diff
@@ -10,7 +10,11 @@ RUN apk update && \
         mkdir -p /var/jenkins_home/init.groovy.d && \
         git clone https://github.com/awslabs/git-secrets.git && \
         make install -C git-secrets && \
-        git-secrets --register-aws --global
+        git-secrets --register-aws --global && \
+#       Additional git secrets rule to check for AWS account numbers (any set of 12 digits)
+        git-secrets --add --global '([^0-9])*[0-9]{12}([^0-9])*' && \
```

Comment:
I was just thinking about this regex - won't it catch longer numbers as well? For example "1234567890123456789" would still match.

Reply:
Playing about in https://regex101.com/, `(^|[^0-9])+[0-9]{12}($|[^0-9])+` looks like it would work, but it's a bit complicated. The first `^` means "start of the line", and `$` means "end of the line". So this expression means "a 12 digit number that's preceded by a non-number _or_ the start of the line, and is followed by a non-number _or_ the end of the line".

I bet there's something simpler, though.

Reply:
Actually, you don't need the `+`, because you don't really care how many characters are before or after the number, as long as there's at least one. So this is slightly simpler, and I think it also works: `(^|[^0-9])[0-9]{12}($|[^0-9])`.

Reply:
No because of the exclusion rule at the beginning and end: ([^0-9])* (ie 0 or more of any character except arabic numerals 0-9). 

I ran these rule against a test file locally with the following combinations in it:

```
This is a test
  123456789012
  768792374148
C768792374148
768792374148 somestring
somestring768792374148somestring
768792374148somestring
somestring5768792374148

12343543657657647568776987697809808909

  $%:!768792374148^&*()
123456789012 somestring
somestring123456789012somestring
123456789012somestring
somestring123456789012


```
Giving the result:
```
[tking@localhost ~]$ git-secrets --scan './gitsecrets_test'

./gitsecrets_test:3:  768792374148
./gitsecrets_test:4:C768792374148
./gitsecrets_test:5:768792374148 somestring
./gitsecrets_test:6:somestring768792374148somestring
./gitsecrets_test:7:768792374148somestring
./gitsecrets_test:12:  $%:!768792374148^&*()

[ERROR] Matched one or more prohibited patterns

```
I've tested it against 31 TDR repos aswell, and didn't see any false positives.

Reply:
As discussed, it looks like git-secrets already does search by whole words, so `\d{12}` would only match strings of 12 digits and ignore longer ones, but this more complicated regex might catch more cases, such as if an account number was concatenated with a string. This might be overkill, but we'll try it out to see if catches anything.